tests/e2e/volumeproperty_test.go

```diff
@@ -0,0 +1,182 @@
+	for _, volumeName := range s.volumeNames {
+		s.getVolumePropertiesDockerCli(c, volumeName)
+	}
+
+	// Verifying docker cli and admin cli properties of a volume are same
+	status := reflect.DeepEqual(dockerCliMap, adminCliMap)
+	c.Assert(status, Equals, true, Commentf("Admin cli and docker cli properties of volumes are not same"))
+
+	// Again verifying if admin cli properties of a volume are same as the expected values
+	s.verifyProperties(c)
+
+	// attaching only thin volume
+	out, err := dockercli.AttachVolume(s.dockerHostIP, s.volumeNames[0], s.containerName)
+	c.Assert(err, IsNil, Commentf("Failed to attach the volume [%s]", out))
+	s.volumeAttached = true
+
+	// As "thin" volume is now attached, again getting properties using admin cli and docker cli
+	s.getVolumePropertiesAdminCli(c)
+	s.getVolumePropertiesDockerCli(c, s.volumeNames[0])
+
+	// Again verifying docker cli and admin cli properties of a volume are same
+	status = reflect.DeepEqual(dockerCliMap, adminCliMap)
+	c.Assert(status, Equals, true, Commentf("Admin cli and docker cli properties of volumes are not same"))
+
+	// Again verifying if admin cli properties of a volume are same as the expected values
+	s.verifyProperties(c)
+
+	log.Printf("END: volumeproperty_test.TestVolumeProperties")
+}
+
+// createVolumes - creates volumes of each format type
+func (s *VolumePropertyTestSuite) createVolumes(c *C) {
+	for i, formatType := range s.formatTypes {
+		s.volumeNames[i] = inputparams.GetVolumeNameWithTimeStamp(formatType)
+		out, err := dockercli.CreateVolumeWithOptions(s.dockerHostIP, s.volumeNames[i], diskFormatOption+formatType)
+		c.Assert(err, IsNil, Commentf(out))
+
+		// creating a single cmd to remove all the created volumes.
+		// Cmd will be used in the test clean-up. e.g. docker volume rm vol1 vol2 vol3
+		s.dockerVolumeRemoveCmd = s.dockerVolumeRemoveCmd + " " + s.volumeNames[i]
+	}
+}
+
+// getPropertiesAndVerify - gets properties of a volume from esx
+func (s *VolumePropertyTestSuite) getVolumePropertiesAdminCli(c *C) {
+	adminCliMap = verification.GetVolumePropertiesAdminCli(s.esxIP)
+	c.Assert(len(adminCliMap), Not(Equals), 0, Commentf("Map of volume names and their corresponding properties is EMPTY."))
+}
+
+// getVolumePropertiesDockerCli - gets properties of a volume from the docker host
+func (s *VolumePropertyTestSuite) getVolumePropertiesDockerCli(c *C, volumeName string) {
+	out, err := verification.GetVolumePropertiesDockerCli(volumeName, s.dockerHostIP)
+	c.Assert(err, IsNil, Commentf(out))
+
+	dockerCliValues := strings.Fields(out)
+	dockerCliMap[volumeName] = [3]string{dockerCliValues[0], dockerCliValues[1], dockerCliValues[2]}
+}
+
+// verifyProperties - verifies admin cli values for a volume are same as the expected values
+// Only for 'thin', after volume is attached, modifying attached-to-vm field with vm name.
+func (s *VolumePropertyTestSuite) verifyProperties(c *C) {
+	for i, volumeName := range s.volumeNames {
+		if s.volumeAttached {
```

Comment:
This is common for all volumes and hence when thin disk is attached, we will always go thru this path and mark all with the docker host name like as though they are attached, which isn't the case.